stumpy/core.py

```diff
@@ -589,6 +594,41 @@ def check_window_size(m, max_size=None):
     if max_size is not None and m > max_size:
         raise ValueError(f"The window size must be less than or equal to {max_size}")
 
+    if n is not None:
+        # Raise warning if there is at least one subsequence with no
+        # non-trivial neighbour in the case of a self-join.
+
+        # For any time series `T`, an "eligible nearest neighbor" subsequence for
+        # the central-most subsequence must be located outside the `excl_zone`,
+        # and the central-most subsequence will ALWAYS have the smallest index-wise
+        # distance to its furthest neighbour amongs all other subsequences. Therefore,
```

Comment:
@seanlaw 

Regarding: 

> this central-most subsequence will ALWAYS have the smallest number of "eligible nearest neighbors" amongst all other subsequences. 

All subsequences with a start index that is at least `excl_zone` indices away from index `0` and `len(T) - m`  have the same number of neighbours. The central-most subsequence, however, has the smallest value for `max(n_left_neighbours, n_right_neighbours)`.

To address this, I've changed `the smallest number of "eligible nearest neighbors"` to `the smallest index-wise distance to its furthest neighbour`

---

Do you know a better/clearer phrase instead of `index-wise distance`?

Reply:
I think "the smallest relative (index-wise) distance to its farthest neighbor"

Also, "amongst" is missing a "t"